package/utils/busybox/config/editors/Config.in

```diff
@@ -128,6 +128,7 @@ config BUSYBOX_CONFIG_FEATURE_VI_REGEX_SEARCH
 	bool "Enable regex in search and replace"
 	default BUSYBOX_DEFAULT_FEATURE_VI_REGEX_SEARCH   # Uses GNU regex, which may be unavailable. FIXME
 	depends on BUSYBOX_CONFIG_FEATURE_VI_SEARCH
+	depends on CONFIG_USE_GLIBC   # musl libc does not support GNU regex
```

Comment:
@satmandu this should be `USE_GLIBC`, right?

Reply:
As in 9d156dd231860767a76ec5020496af197e8fc346

Reply:
If I recalll correctly, CONFIG_USE_GLIBC shows up through various Makefiles too, so I was unsure as to which to use. The CONFIG_USE_GLIBC depend did hide the option, but I'm unsure which is the appropriate one to use.

Example in ```package/system/ubox/Makefile```:

```
TARGET_LDFLAGS += $(if $(CONFIG_USE_GLIBC),-lrt)
```

Reply:
Are you able to select `BUSYBOX_DEFAULT_FEATURE_VI_REGEX_SEARCH` option with your patch and GNU libc?

Reply:
You are correct. I was in error here. The depend should be upon USE_GLIBC for ```BUSYBOX_DEFAULT_FEATURE_VI_REGEX_SEARCH ``` to be an option with GLIBC enabled. 

I can confirm that ```BUSYBOX_DEFAULT_FEATURE_VI_REGEX_SEARCH ``` IS an option if one switches to glibc from musl.

Reply:
```
 .config - OpenWrt Configuration                                                 
 [...] ............................ Core utilities for embedded Linux > Editors  
  lqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq Editors qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqk   
  x  Arrow keys navigate the menu.  <Enter> selects submenus ---> (or empty  x   
  x  submenus ----).  Highlighted letters are hotkeys.  Pressing <Y>         x   
  x  includes, <N> excludes, <M> modularizes features.  Press <Esc><Esc> to  x   
  x  exit, <?> for Help, </> for Search.  Legend: [*] built-in  [ ] excluded x   
  x lqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqk x   
  x x    [*] awk (23 kb)                                                   x x   
  x x    [*]   Enable math functions (requires libm)                       x x   
  x x    [*]   Enable a few GNU extensions                                 x x   
  x x    [*] cmp (4.9 kb)                                                  x x   
  x x    [*] diff (13 kb)                                                  x x   
  x x    [ ]   Enable long options                                         x x   
  x x    [*]   Enable directory support                                    x x   
  x x    [ ] ed (21 kb)                                                    x x   
  x x    [*] patch (9.4 kb)                                                x x   
  x x    [*] sed (12 kb)                                                   x x   
  x x    [*] vi (23 kb)                                                    x x   
  x x    (1024) Maximum screen width                                       x x   
  x x    [ ]   Allow to display 8-bit chars (otherwise shows dots)         x x   
  x x    [*]   Enable ":" colon commands (no "ex" mode)                    x x   
  x x    [*]   Enable yank/put commands and mark cmds                      x x   
  x x    [*]   Enable search and replace cmds                              x x   
  x x    [ ]     Enable regex in search and replace (NEW)                  x x 
```
```
 .config - OpenWrt Configuration                                                 
 [...] ............................ Core utilities for embedded Linux > Editors  
  lqqqqqqqqqqqqqqqqqqq Enable regex in search and replace qqqqqqqqqqqqqqqqqqqk   
  x CONFIG_BUSYBOX_CONFIG_FEATURE_VI_REGEX_SEARCH:                           x   
  x                                                                          x   
  x Use extended regex search.                                               x   
  x                                                                          x   
  x Symbol: BUSYBOX_CONFIG_FEATURE_VI_REGEX_SEARCH [=n]                      x   
  x Type  : boolean                                                          x   
  x Prompt: Enable regex in search and replace                               x   
  x   Location:                                                              x   
  x     -> Base system                                                       x   
  x       -> busybox................................ Core utilities for embe x   
  x         -> Editors                                                       x   
  x           -> vi (23 kb) (BUSYBOX_CONFIG_VI [=y])                         x   
  x             -> Enable search and replace cmds (BUSYBOX_CONFIG_FEATURE_VI x   
  x   Defined at package/utils/busybox/config/editors/Config.in:127          x   
  x   Depends on: PACKAGE_busybox [=y] && BUSYBOX_CUSTOM [=y] && \           x   
  x BUSYBOX_CONFIG_FEATURE_VI_SEARCH [=y] && USE_GLIBC [=y]                  x 
```

Reply:
Wit musl-lib enabled:
```
 .config - OpenWrt Configuration                                                 
 [...] ............................ Core utilities for embedded Linux > Editors  
  lqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq Editors qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqk   
  x  Arrow keys navigate the menu.  <Enter> selects submenus ---> (or empty  x   
  x  submenus ----).  Highlighted letters are hotkeys.  Pressing <Y>         x   
  x  includes, <N> excludes, <M> modularizes features.  Press <Esc><Esc> to  x   
  x  exit, <?> for Help, </> for Search.  Legend: [*] built-in  [ ] excluded x   
  x lqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqk x   
  x x    [*] awk (23 kb)                                                   x x   
  x x    [*]   Enable math functions (requires libm)                       x x   
  x x    [*]   Enable a few GNU extensions                                 x x   
  x x    [*] cmp (4.9 kb)                                                  x x   
  x x    [*] diff (13 kb)                                                  x x   
  x x    [ ]   Enable long options                                         x x   
  x x    [*]   Enable directory support                                    x x   
  x x    [ ] ed (21 kb)                                                    x x   
  x x    [*] patch (9.4 kb)                                                x x   
  x x    [*] sed (12 kb)                                                   x x   
  x x    [*] vi (23 kb)                                                    x x   
  x x    (1024) Maximum screen width                                       x x   
  x x    [ ]   Allow to display 8-bit chars (otherwise shows dots)         x x   
  x x    [*]   Enable ":" colon commands (no "ex" mode)                    x x   
  x x    [*]   Enable yank/put commands and mark cmds                      x x   
  x x    [*]   Enable search and replace cmds                              x x   
  x x    [*]   Catch signals                                               x x
```

Reply:
Ok, thanks for confirmation.

Reply:
Updated patch thus: https://github.com/satmandu/openwrt/commit/f92f86c47d77960ddfe05104b7d39f3021ddf353